HPCC-16877 Xref not showing found file

miguelvazq opened the pull request:
Found file not displaying when user browses the found file tab.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

miguelvazq commented:
@GordonSmith Please review.

A few things are happening here: 

- When the response only contains one file it is returned in a JSONObject when it is multiple rows it is returned in a JSONArray so I have added logic to handle both scenarios.
- Attach functionality was added
- I have used the dojo/all to handle the removal of multiple files
- Added selector to dgrid

HPCCSmoketest commented:
Automated Smoketest
Sha: 600908ad2c9666b77929f9d982f16d07b9527fdc
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.146 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith commented on esp/src/eclwatch/XrefFoundFilesWidget.js:
```diff
@@ -70,6 +98,57 @@ define([
         },
 
+        refreshActionState: function (event) {
+            var context = this;
```
Context not used.

GordonSmith commented on esp/src/eclwatch/XrefFoundFilesWidget.js:
```diff
@@ -82,15 +161,27 @@ define([
+                            Modified: results.Modified,
+                            Parts: results.Numparts,
+                            Size: results.Size
+                        })
```
No need for the mixin, since the target is empty.

miguelvazq replied:
If there is a file present (single row) we need to manually insert the row into an array.

GordonSmith replied:
The following code is equivilant:
```
newRows.push({
    Name: results.Partmask,
    Modified: results.Modified,
    Parts: results.Numparts,
    Size: results.Size
});
```

GordonSmith commented on esp/src/eclwatch/XrefLostFilesWidget.js:
```diff
@@ -89,19 +135,34 @@ define([
+                            Partslost: results.Partslost,
+                            Primarylost: results.Primarylost,
+                            Replicatedlost: results.Replicatedlost
+                        })
```
No need for the mixin, since the target is empty. (and missing ";")

GordonSmith commented on esp/src/eclwatch/XrefOrphanFilesWidget.js:
```diff
@@ -112,16 +136,29 @@ define([
+                            PartsFound: results.Partsfound,
+                            TotalParts: results.Numparts,
+                            Size: results.Size
+                        })
```
No need for the mixin, since the target is empty. (and missing ";")

GordonSmith commented on esp/src/eclwatch/XrefQueryWidget.js:
```diff
@@ -183,13 +183,25 @@ define([
+                        Name: results.Name,
+                        Modified: results.Modified,
+                        Status: results.Status
+                    })
```
No need for the mixin, since the target is empty.

richardkchapman commented:
@miguelvazq Back to you

miguelvazq commented:
Updated.

HPCCSmoketest commented:
Automated Smoketest
Sha: 578d3b4cf3da0469c96454b88e92c55065a84153
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.509 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith commented:
Looks fine.

miguelvazq commented:
Verified by @cloLN Errors/Warnings has same issue. Let me add same logic.

miguelvazq commented:
@GordonSmith added logic to errors/warnings widget to handle this same scenario. FYI @cloLN

Commit:
HPCC-16877 Xref not showing found file

Found file not displaying when user browses the found file tab.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest
Sha: 6cca2db26a5d32cc3e12d70de95d7808cc685edd
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.23 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest
Sha: 7db9c9ed41aa5f85548f532cd921c4f0e06f75c3
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.688 seconds |

Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unittest result:

| total | passed | failed | timeout | 
|---|---|---|---|
|86 |86 |0 |0 |

HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith commented on esp/src/eclwatch/XrefErrorsWarningsWidget.js:
```diff
@@ -97,22 +97,38 @@ define([
+                            Status: context.i18n.Error
+                        });
+                    }
+                }
```
This should be more along the lines of:
```
if (lang.exists("Warning.length", results)) {
	arrayUtil.forEach(results.Warning, function (row, idx) {
	   newRows.push({
			File: row.File,
			Text: row.Text,
			Status: context.i18n.Warning
		});
	});
} else if (results.Warning) {
	newRows.push({
		File: results.Warning.File,
		Text: results.Warning.Text,
		Status: context.i18n.Warning
	});
}

if (lang.exists("Error.length", results)) {
	arrayUtil.forEach(results.Error, function (row, idx) {
	   newRows.push({
			File: row.File,
			Text: row.Text,
			Status: context.i18n.Error
		});
	});
} else if (results.Error) {
	newRows.push({
		File: results.Error.File,
		Text: results.Error.Text,
		Status: context.i18n.Error
	});
}
```

GordonSmith commented on esp/src/eclwatch/XrefFoundFilesWidget.js:
```diff
@@ -70,6 +98,56 @@ define([
+        _onDeleteFiles: function (event) {
+            var context = this;
+            var selections = this.grid.getSelected();
+            if (confirm(this.i18n.DeleteSelectedFiles)) {
```
Should display a list of the items you are about to delete (similar to other places we delete)

GordonSmith commented on esp/src/eclwatch/XrefFoundFilesWidget.js:
```diff
@@ -70,6 +98,56 @@ define([
+        _onAttachFiles: function (event) {
+            var context = this;
+            var selections = this.grid.getSelected();
+            if (confirm(this.i18n.AddTheseFilesToDali)) {
```
hould display a list of the items you are about to delete (similar to other places we delete)

GordonSmith commented on esp/src/eclwatch/XrefFoundFilesWidget.js:
```diff
@@ -70,6 +98,56 @@ define([
+                            Cluster: context.params.Name,
+                            Type: 'Found',
+                            Action: 'Delete',
+                            XRefFiles_i2: row.Name
```
Should probably be "XRefFiles_i0"?

In fact you can probably group all the actions into a single call using XRefFiles_i0, XRefFiles_i1, XRefFiles_i2...  (search for "flattenArray" for other places we do this.)

GordonSmith commented on esp/src/eclwatch/XrefFoundFilesWidget.js:
```diff
@@ -70,6 +98,56 @@ define([
+                            Cluster: context.params.Name,
+                            Type: 'Found',
+                            Action: 'Attach',
+                            XRefFiles_i2: row.Name
```
See previous comment re _i0

GordonSmith commented on esp/src/eclwatch/XrefLostFilesWidget.js:
```diff
@@ -78,6 +95,34 @@ define([
+                            Cluster: context.params.Name,
+                            Type: 'Lost',
+                            Action: 'DeleteLogical',
+                            XRefFiles_i2: row.Name
```
See previous comment re _i0

GordonSmith commented on esp/src/eclwatch/XrefLostFilesWidget.js:
```diff
@@ -89,19 +134,32 @@ define([
+                            Primarylost: results.Primarylost,
+                            Replicatedlost: results.Replicatedlost
+                        });
+                    }
```
Note:  Having to check the response structures from ESP for the existence of an array smells like a bug in the ESP API (which may be too late to do anything about).

I would suggest moving this "cleanup" logic into WsDFUXref.

GordonSmith commented on esp/src/eclwatch/XrefOrphanFilesWidget.js:
```diff
@@ -100,6 +95,34 @@ define([
+        _onDeleteFiles: function (event) {
+            var context = this;
+            var selections = this.grid.getSelected();
+            if (confirm(this.i18n.DeleteSelectedFiles)) {
```
Confirmation should include list of items

GordonSmith commented on esp/src/eclwatch/XrefOrphanFilesWidget.js:
```diff
@@ -100,6 +95,34 @@ define([
+                            Cluster: context.params.Name,
+                            Type: 'Orphan',
+                            Action: 'Delete',
+                            XRefFiles_i2: row.Name
```
See previous comment re _i0

richardkchapman commented:
@miguelvazq Back to you

miguelvazq commented:
Cherry-picked and tested in emulation mode. IE8 is successfully displaying treemap and a graph on the following page. Looks good.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 7d03b27237e1a66cffee1ad6f4bef4f155ba36a0
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 1 | 65 | 89.43 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

miguelvazq commented:
@GordonSmith Changes made.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: dca057ac3af26ba6a0465d53ffac16ca6c61ed3a
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 1 | 65 | 91.522 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: dca057ac3af26ba6a0465d53ffac16ca6c61ed3a
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 1 | 65 | 90.376 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith commented on esp/src/eclwatch/XrefFoundFilesWidget.js:
```diff
@@ -29,13 +29,16 @@ define([
     "hpcc/WsDFUXref",
     "hpcc/DelayLoadWidget",
-    "hpcc/ESPUtil"
+    "hpcc/ESPUtil",
```
Trailing comma

GordonSmith reviewed (changes requested):
There is a build error - I suspect the trailing comma below is the culprit.  I will do a full review once that is resolved.

miguelvazq commented:
@GordonSmith modified. Waiting for build passing

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 32334c4cd0e948ee03c5ed266ca1bbc3c7b4eadc
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.858 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |86 |86 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith commented on esp/src/eclwatch/_Widget.js:
```diff
@@ -286,15 +286,24 @@ define([
         arrayToList: function (arr, field) {
             var retVal = "";
-            arrayUtil.some(arr, function (item, idx) {
+             arrayUtil.some(arr, function (item, idx) {
```
Minor:  indentation

GordonSmith commented on esp/src/eclwatch/_Widget.js:
```diff
@@ -286,15 +286,24 @@ define([
+                        return true;
+                    }
+                    retVal += field ? item[field] : item;
                 }
```
Duplicate code - would suggest something like:
```
...
if (idx >= 10) {
    retVal += "\n..." + (arr.length - 10) + " " + this.i18n.More + "...";
    return true;
}
var lineStr = field ? item[field] : item;
if (lineStr.length > 50) {
    retVal += "..." + lineStr.slice(25,item[field].length);
} else {
    retVal += lineStr;
}
...
```

GordonSmith reviewed (changes requested):
Minor indentations and some duplicate code.

richardkchapman commented:
@miguelvazq Back to you

miguelvazq commented:
@GordonSmith changes made.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 26302aa51be1394d5ce2c00823e2f55172e7c69a
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.839 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



HPCC Stop: OK
HPCC Uninstall: OK

GordonSmith reviewed (approved):
Looks good.
